blog/authenticating_a_rails_api_part_2.md

```diff
@@ -0,0 +1,184 @@
+## So you want to sign up users and authenticate your Rails API - Part 2
+
+**Note:** This tutorial is designed for someone who has a beginning to intermediate knowledge of Ruby on Rails.
+
+In our last post, we reviewed signing up users via a `POST /users` endpoint in our API.  In this episode, we'll review how we authenticate users using an authentication token passed in the authorization header.
+
+### Overview
+
+We use [warden](https://github.com/hassox/warden) to authenticate our endpoints.  Warden lets you define custom strategies for restricting access to routes in your application. This is especially useful if you have different authentication rules for different endpoints.
+
+In this post, we'll create a `GET /widgets` endpoint that can only be accessed by our users.  We'll require users to pass up their authentication token in an authorization header like so:
+
+```
+GET /widgets
+Authorization: Token token=<my_token_here>
```

Comment:
Occasionally the mobile devs have asked why it has to have the whole `Token token=` thing instead of just the token itself. Might want to try to find a citation for why that is an agreed upon prefix.